azurerm/internal/services/timeseriesinsights/resource_time_series_insights_environment.go

```diff
@@ -0,0 +1,293 @@
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("retrieving Time Series Insights Environment %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resource, ok := resp.Value.AsStandardEnvironmentResource()
+	if !ok {
+		return fmt.Errorf("resource was not a standard Time Series Insights Environment %q (Resource Group %q)", name, resourceGroup)
+	}
+
+	if resource.ID == nil {
+		return fmt.Errorf("cannot read Time Series Insights Environment %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*resource.ID)
+
+	return resourceArmTimeSeriesInsightsEnvironmentRead(d, meta)
+}
+
+func resourceArmTimeSeriesInsightsEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).TimeSeriesInsights.EnvironmentsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.TimeSeriesInsightsEnvironmentID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
+	if err != nil || resp.Value == nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Time Series Insights Environment %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	environment, ok := resp.Value.AsStandardEnvironmentResource()
+	if !ok {
+		return fmt.Errorf("exisiting resource was not a standard Time Series Insights Environment %q (Resource Group %q)", id.Name, id.ResourceGroup)
+	}
+
+	d.Set("name", environment.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("sku_name", flattenEnvironmentSkuName(environment.Sku))
+	if location := environment.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := environment.StandardEnvironmentResourceProperties; props != nil {
+		d.Set("storage_limited_exceeded_behavior", string(props.StorageLimitExceededBehavior))
+		d.Set("data_retention_time", props.DataRetentionTime)
+
+		if partition := props.PartitionKeyProperties; partition != nil && len(*partition) > 0 {
+			for _, v := range *partition {
+				d.Set("partition_key", *v.Name)
+			}
+		}
+	}
+
+	return tags.FlattenAndSet(d, environment.Tags)
+}
+
+func resourceArmTimeSeriesInsightsEnvironmentDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).TimeSeriesInsights.EnvironmentsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.TimeSeriesInsightsEnvironmentID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	response, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(response) {
+			return fmt.Errorf("deleting Time Series Insights Environment %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
```

Comment:
I suppose if the service returns 404 when you are deleting the resource, it is still something wrong happening. Typically if you are attempting to delete a non-existing resource, the service should return 204